src/Sql/dbo/Stored Procedures/ProviderUser_ReadCountByProviderIdEmail.sql

```diff
@@ -0,0 +1,21 @@
+﻿CREATE PROCEDURE [dbo].[ProviderUser_ReadCountByProviderIdEmail]
+    @ProviderId UNIQUEIDENTIFIER,
+    @Email NVARCHAR(256),
+    @OnlyUsers BIT
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SELECT
+        COUNT(1)
+    FROM
+        [dbo].[ProviderUser] OU
+    LEFT JOIN
+        [dbo].[User] U ON OU.[UserId] = U.[Id]
+    WHERE
+        OU.[ProviderId] = @ProviderId
+        AND (
+            (@OnlyUsers = 0 AND (OU.[Email] = @Email OR U.[Email] = @Email))
```

Comment:
Same, `OR` will have a negative impact on performance... use the `IN` operator instead where applicable:
```sql
(@OnlyUsers = 0 AND @Email IN (OU.[Email], U.[Email]))
```